sdk/identity/azure-identity/TOKEN_CACHING.md

```diff
@@ -0,0 +1,76 @@
+## Token caching in the Azure Identity client library
+
+*Token caching* is a feature provided by the Azure Identity library that allows apps to:
+
+- Improve their resilience and performance.
+- Reduce the number of requests made to Azure Active Directory (Azure AD) to obtain access tokens.
+  
+When an app needs to access a protected Azure resource, it typically needs to obtain an access token from Azure AD. Obtaining that token involves sending a request to Azure AD. Azure AD then validates the user's credentials and issues an access token.
+
+Token caching, via the Azure Identity library, allows the app to store this access token [in memory](#in-memory-token-caching) or [on disk](#persistent-token-caching). The token can then be retrieved quickly and easily the next time the app needs to access the same resource. The app can avoid making another request to Azure AD, which reduces network traffic and improves resilience.
+
+### In-memory token caching
+
+*In-memory token caching* is the default option provided by the Azure Identity library. This caching approach allows apps to store access tokens in memory. With in-memory token caching enabled, the library first determines if a valid access token for the requested resource is already stored in memory. If a valid token is found, it's returned to the app without the need to make another request to Azure AD.
+
+The in-memory token cache provided by the Azure Identity library:
+
+- Is thread-safe.
+- Can be used by multiple threads concurrently.
+
+**Note:** When Azure Identity library credentials are used with Azure service libraries (for example, Azure Blob Storage), the in-memory token caching is active in the `HttpPipeline` layer as well. All `TokenCredential` implementations are supported there, including custom implementations external to the Azure Identity library.
+
+### Persistent token caching
+
+*Persistent disk token caching* is an opt-in feature in the Azure Identity library. The feature allows apps to cache access tokens in an encrypted, persistent storage mechanism. As indicated in the following table, the storage mechanism differs across operating systems.
```

Comment:
Should this also indicate that the file can be stored unencrypted using `allowUnencryptedStorage`? If we do mention it should probably be after the table of storage mechanisms, and we should also note that it is substantially less secure, as tokens are not encrypted to the current user so others with access to the machine may also be able to access tokens from the cache.

Reply:
Given that its less secure and leaves tokens exposed, leaning towards not promoting it here.
@scottaddie any thoughts ?